sensor_core/dsp/dsp_manager.py

```diff
@@ -3,59 +3,87 @@
 import scipy.signal as signal
 
 class DSPManager(DictManager):
-    def _init_(self, dsp_module_queue: np.array([]), dsp_modules: dict = None):
+    def _init_(self, dsp_modules_queue: np.array([]), dsp_modules: dict = None):
         """Initialize empty array and create empty dictionary
         :param dsp_modules: Array of dsp_module objects, identified by user-defined name, target algo, and key parameters
         :param dsp_dict: master dsp_module dictionary
         :param dict_type: Dynamic dictionary """
 
         self.dsp_modules = {}
-        self.dsp_module_queue = []
+        self.dsp_modules_queue = []
 
+    @staticmethod
     def add_dsp_module(self, name: str, target_algo: str, **kwargs):
-        ma_keys = {'N': int, 'pad': str}
+        #TODO: Make sure no duplicate names, 
+        # Make sure input parameters are correct data type
+        # Currently only checks after run_dsp_module executes
+
+        ma_keys = {'window_size': int, 'pad': str}
         bw_keys = {'order': int, 'min_frq': int, 'max_frq': int, 'btype': str}
 
-        if (target_algo == 'moving_average_filter'):
+        if target_algo == 'moving_average_filter':
             if not set(ma_keys).issubset(kwargs.keys()):
                 raise ValueError(f"Missing required parameters: {ma_keys}")
```

Comment:
Might be good to explicitly list what parameters are missing, rather than just saying what params they need?